### PR TITLE
Fix a problem with aromatic heteroatom tautomer enumeration

### PR DESCRIPTION
**Pull request by greglandrum**
There was a problem with moving Hs onto/off of charged aromatic N atoms.
This updates the transform rules so that that no longer happens.

Fixes one of the problems raised in #2908

### CODE REVIEW

**Comment by greglandrum** [comment]
@mcs07 : if you have time to take a look a this, I'd love your comments.
The transforms we are using came from molvs originally, so you might find the changes useful there too.

File: Code/GraphMol/MolStandardize/Tautomer.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include <GraphMol/MolStandardize/FragmentCatalog/FragmentCatalogUtils.h>
 #include <GraphMol/SmilesParse/SmilesParse.h>
 #include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <GraphMol/SmilesParse/SmartsWrite.h>
```
Comment (bp-kelley) [inline Code/GraphMol/MolStandardize/Tautomer.cpp]: It looks like this isn’t really used ( moltosmarts is commented out )

Comment (greglandrum) [reply Code/GraphMol/MolStandardize/Tautomer.cpp]: good catch. fixed

**Comment by mcs07** [comment]
Hi Greg, this seems reasonable to me, although it has been a long while since I thought deeply about this. At least it doesn't break anything in the small collection of tests in MolVS.

To be honest, I've always felt that this method of enumerating via transforms and scoring to find a canonical tautomer is a bit flawed, and it will always be a struggle to make it robust and efficient. For ages I've been meaning to try do an implementation that is closer to the [Sayle/Delany method](https://www.daylight.com/meetings/emug99/Delany/taut_html/sld001.htm) but never got around to it...

**Comment by greglandrum** [comment]
Thanks @mcs07!
I agree that the Sayle/Delany approach is also interesting (and something I may also look at as I continue to fall into this rabbit hole), but the transform-based approach is still certainly useful and has the advantage of being pretty easy to explain and expand.

**Pull request merged**